src/ts/components/NavigatorComponent.ts

```diff
@@ -0,0 +1,103 @@
+import ProductInputComponent from './ProductManageComponent/ProductInputComponent';
+import ProductStateComponent from './ProductManageComponent/ProductsStateComponent';
+import CoinInputComponent from './CoinManageComponent/CoinInputComponent';
+
+import VendingMachineProductManager from '../domains/VendingMachineProductManager';
+import VendingMachineCoinManager from '../domains/VendingMachineCoinManager';
+import CoinsStateComponent from './CoinManageComponent/CoinsStateComponent';
+
+import { $, on } from '../dom';
+import { ROUTES } from '../constants';
+
+export default class NavigatorComponent {
+  private $productInfoSection: HTMLElement = $('.product-info-section');
+  private $chargeCoinSection: HTMLElement = $('.charge-coin-section');
+  private $navProductButton = $('.nav__product-button') as HTMLButtonElement;
+  private $navChargeButton = $('.nav__charge-button') as HTMLButtonElement;
+  private $coinInput = $(
+    '.charge-form-section__coin-input'
+  ) as HTMLInputElement;
+  private $productInput = $(
+    '.product-info-form__product-input'
+  ) as HTMLInputElement;
+
+  private vendingMachineProductManager = new VendingMachineProductManager();
+  private vendingMachineCoinManager = new VendingMachineCoinManager();
+
+  constructor() {
+    new ProductStateComponent(this.vendingMachineProductManager);
+    new ProductInputComponent(this.vendingMachineProductManager);
+    new CoinInputComponent(this.vendingMachineCoinManager);
+    new CoinsStateComponent();
+
+    on(this.$navProductButton, 'click', this.onClickNavProductButton);
+    on(this.$navChargeButton, 'click', this.onClickNavChargeButton);
+    on(window, 'popstate', this.onPopstateRoute);
+    this.routeURLVisit(window.location.pathname);
+  }
+
+  private routeURLVisit(pathname: string): void {
+    if (
+      !Object.values(ROUTES).some((route) => route === window.location.pathname)
+    ) {
+      window.history.replaceState(null, null, '/');
+
+      return;
+    }
+
+    if (pathname === ROUTES.COINS) {
+      this.renderCoinComponent();
+      window.history.pushState({}, '', ROUTES.COINS);
+
+      return;
+    }
+
+    if (pathname === ROUTES.PRODUCTS) {
+      this.renderProductComponent();
+      window.history.pushState({}, '', ROUTES.PRODUCTS);
+
+      return;
+    }
+  }
+
+  private onPopstateRoute = (): void => {
+    if (window.location.pathname === ROUTES.COINS) {
+      this.renderCoinComponent();
+    }
+
+    if (
+      window.location.pathname === ROUTES.PRODUCTS ||
+      window.location.pathname === '/'
```

Comment:
작성해주신 상수활용하면 좋을 것 같구요! 개인적으론 ROUTES.PRODUCTS 자체를 '/' 로 정의해도 무방하지 않을까 생각합니다 ;ㅁ;
```suggestion
      window.location.pathname === ROUTES.HOME
```

Reply:
MAIN_PAGE를 만들어서 products를 메인으로 수정했습니다!
```jsx
 const MAIN_PAGE = 'products';
```